### DISCUSSION TITLE
Is there a way I can adjust the number of hazards in Goal1 environment?

### DISCUSSION THREAD

**Discussion by PiyabhumC** [PKU-Alignment/safety-gymnasium#130] [Q&A]
Hello,
I would like to adjust the amount of hazards appear in the environment Goal1. Ideally I would like to gradually increase the number of hazards as the training progress (increasing difficulties). Is there a way to config this?
Thank you very much.

**Comment by muchvo** [comment]
There are two methods:

1. You can dynamically modify the `num` attribute of `hazards` in the Goal1 task instance within the code, and then call the `reset` method. Refer to [this line of code](https://github.com/PKU-Alignment/safety-gymnasium/blob/main/safety_gymnasium/tasks/safe_navigation/goal/goal_level1.py#L33).  
   Example code:  
   ```python
   env = safety_gymnasium.make(env_name, render_mode='human')
   obs, info = env.reset()  # pylint: disable=unused-variable
   env.hazards.num = 100  # Note: Too many hazards may cause issues with environment instantiation, requiring an adjustment to the space size.
   obs, info = env.reset()  # pylint: disable=unused-variable
   ```

2. Alternatively, you can re-instantiate an environment using a `config` approach. Refer to this [issue](https://github.com/PKU-Alignment/safety-gymnasium/issues/98) for more details.

**Reply by PiyabhumC** [reply]
Thanks. I can manipulate the hazards.num within the goal_level1.py however when I modify it as the example with env.hazards.num it returns AttributeError: 'Builder' object has no attribute 'hazards'.
Is there any outer attribute I have to pass through  as well?

**Reply by muchvo** [reply]
Apologies, the first method is currently not supported. Please use the second method instead.
